src/main/java/com/mraof/minestuck/block/HorseClockBlock.java

```diff
@@ -0,0 +1,141 @@
+package com.mraof.minestuck.block;
+
+import com.mraof.minestuck.block.machine.MachineMultiblock;
+import com.mraof.minestuck.block.machine.MultiMachineBlock;
+import com.mraof.minestuck.blockentity.HorseClockBlockEntity;
+import com.mraof.minestuck.blockentity.MSBlockEntityTypes;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.EntityBlock;
+import net.minecraft.world.level.block.RenderShape;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.entity.BlockEntityTicker;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.block.state.properties.BlockStateProperties;
+import net.minecraft.world.level.block.state.properties.IntegerProperty;
+import net.minecraft.world.phys.BlockHitResult;
+
+import javax.annotation.Nullable;
+import java.util.Random;
+
+public class HorseClockBlock extends MultiMachineBlock implements EntityBlock
+{
+	public static final IntegerProperty POWER = BlockStateProperties.POWER;
+	
+	public HorseClockBlock(MachineMultiblock machine, Properties properties)
+	{
+		super(machine, properties);
+		registerDefaultState(stateDefinition.any().setValue(POWER, 0));
+	}
+	
+	@Override
+	public InteractionResult use(BlockState state, Level level, BlockPos pos, Player player, InteractionHand hand, BlockHitResult hit)
+	{
+		//will only work on the bottom block
+		if(level.getBlockEntity(pos) instanceof HorseClockBlockEntity be)
+		{
+			HorseClockBlockEntity.fullPower(level, be);
+			
+			return InteractionResult.SUCCESS;
+		}
+		
+		return InteractionResult.PASS;
+	}
+	
+	@Nullable
+	@Override
+	public BlockEntity newBlockEntity(BlockPos pos, BlockState state)
+	{
+		if(state.is(MSBlocks.HORSE_CLOCK.BOTTOM.get())) //prevents non block entity blocks from trying to tick
+		{
+			return new HorseClockBlockEntity(pos, state);
+		} else
+			return null;
```

Comment:
I think I recall some sort of minor problem when returning a null block entity. Perhaps it is relevant to avoid this with a new class for the block entity/ticker block, akin to other multiblocks?

Reply:
Created a class for Bottom within HorseClockBlock